src/utils/tests/parser.js

```diff
@@ -96,46 +99,110 @@ describe("parser", () => {
     });
   });
 
-  describe("getPathClosestToLocation", () => {
-    it("Can find the function declaration for square", () => {
-      const closestPath = getPathClosestToLocation(getSourceText("func"), {
-        line: 1,
-        column: 1
+  describe("getClosestExpression", () => {
+    it("Can find a member expression", () => {
+      const expression = getClosestExpression(
+        getSourceText("resolveToken"),
+        "x",
+        {
+          line: 15,
+          column: 31
+        }
+      );
+
+      expect(expression.value).to.be("obj.x");
+      expect(expression.location.start).to.eql({
+        line: 15,
+        column: 26
       });
+    });
 
-      expect(closestPath.node.id.name).to.be("square");
-      expect(closestPath.node.loc.start).to.eql({
-        line: 1,
-        column: 0
+    it("Can find a local var", () => {
+      const expression = getClosestExpression(
+        getSourceText("resolveToken"),
+        "beta",
+        {
+          line: 15,
+          column: 21
+        }
+      );
+
+      expect(expression.value).to.be("beta");
+      expect(expression.location.start).to.eql({
+        line: 15,
+        column: 19
       });
-      expect(closestPath.type).to.be("FunctionDeclaration");
     });
+  });
 
-    it("Can find the path at the exact column", () => {
-      const closestPath = getPathClosestToLocation(getSourceText("func"), {
+  describe("getClosestScope", () => {
+    it("Can find the function declaration for square", () => {
+      const scope = getClosestScope(getSourceText("func"), {
         line: 2,
-        column: 10
+        column: 1
       });
 
-      expect(closestPath.node.loc.start).to.eql({
-        line: 2,
-        column: 9
+      const node = scope.block;
+      expect(node.id.name).to.be("square");
+      expect(node.loc.start).to.eql({
+        line: 1,
+        column: 0
       });
-      expect(closestPath.type).to.be("Identifier");
+      expect(node.type).to.be("FunctionDeclaration");
     });
+  });
 
+  describe("getVariablesInLocalScope", () => {
     it("finds scope binding variables", () => {
-      var vars = getVariablesInScope(getSourceText("math"), {
-        line: 1,
+      const scope = getClosestScope(getSourceText("math"), {
```

Comment:
yeah, so scope is a really big object that i don't really want to serialize. it's big because it has lots of info about parents and basically the tree :/